python/lsst/ctrl/oods/msgIngester.py

```diff
@@ -129,13 +142,17 @@ def run_tasks(self):
         # do the ingest
 
         task = asyncio.create_task(self.dequeue_and_ingest_files())
+        task.add_done_callback(self.helper_done_callback)
         self.tasks.append(task)
 
-        clean_tasks = self.get_butler_clean_tasks()
-        for clean_task in clean_tasks:
-            task = asyncio.create_task(clean_task())
+        butler_tasks = self.get_butler_tasks()
+        for butler_task in butler_tasks:
+            task = asyncio.create_task(butler_task())
+            task.add_done_callback(self.helper_done_callback)
             self.tasks.append(task)
 
+        self.tasks.append(task)
+
```

Comment:
I might be missing something, but won't this line cause the last task in the `butler_tasks` appear twice in the `self.tasks`?

Reply:
Fixed.